app/controllers/stages_controller.rb

```diff
@@ -102,4 +99,12 @@ def stage_params
   def find_stage
     @stage = current_project.stages.find_by_param!(params[:id])
   end
+
+  def template_param
+    @template_param ||= ActiveRecord::Type::Boolean.new.type_cast_from_database(stage_params.delete("template"))
+  end
+
+  def update_template
+    template_param ? @stage.project.template!(@stage) : @stage.project.template!
```

Comment:
why can't it unset the previous stage when `template` is enabled ?

On Tue, Aug 16, 2016 at 2:00 PM, Adam Panzer <notifications@github.com>
wrote:

> In app/controllers/stages_controller.rb
> <https://github.com/zendesk/samson/pull/1198#discussion_r75019104>:
>
> > @@ -102,4 +99,12 @@ def stage_params
> >    def find_stage
> >      @stage = current_project.stages.find_by_param!(params[:id])
> >    end
> > +
> > +  def template_param
> > +    @template_param ||= ActiveRecord::Type::Boolean.new.type_cast_from_database(stage_params.delete("template"))
> > +  end
> > +
> > +  def update_template
> > +    template_param ? @stage.project.template!(@stage) : @stage.project.template!
>
> Also: this can't be an after_save hook because the default behavior will
> still set the passed in stage as template w/o unsetting the previous stage.
>
> —
> You are receiving this because you were mentioned.
> Reply to this email directly, view it on GitHub
> <https://github.com/zendesk/samson/pull/1198/files/d43bdda94cc05d08fa093f7c694b09953335fe89#r75019104>,
> or mute the thread
> <https://github.com/notifications/unsubscribe-auth/AAAsZ9Yk8WIjCIV9Yz09hIjOjewZz8lQks5qgiTdgaJpZM4Jhtog>
> .
>